staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -499,99 +502,22 @@ func (o *ApplyOptions) applyOneObject(info *resource.Info) error {
-				err = fmt.Errorf("Server-side apply not available on the server: (%v)", err)
-			}
-			if errors.IsConflict(err) {
-				err = fmt.Errorf(`%v
-Please review the fields above--they currently have other managers. Here
-are the ways you can resolve this warning:
-* If you intend to manage all of these fields, please re-run the apply
-  command with the `+"`--force-conflicts`"+` flag.
-* If you do not intend to manage all of the fields, please edit your
-  manifest to remove references to the fields that should keep their
-  current managers.
-* You may co-own fields by updating your manifest to match the existing
-  value; in this case, you'll become the manager if the other manager(s)
-  stop managing the field (remove it from their configuration).
-See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`, err)
-			}
-			return err
-		}
-
-		info.Refresh(obj, true)
-
-		// Migrate managed fields if necessary.
-		//
-		// By checking afterward instead of fetching the object beforehand and
-		// unconditionally fetching we can make 3 network requests in the rare
-		// case of migration and 1 request if migration is unnecessary.
-		//
-		// To check beforehand means 2 requests for most operations, and 3
-		// requests in worst case.
-		if err = o.saveLastApplyAnnotationIfNecessary(helper, info); err != nil {
-			fmt.Fprintf(o.ErrOut, warningMigrationLastAppliedFailed, err.Error())
-		} else if performedMigration, err := o.migrateToSSAIfNecessary(helper, info); err != nil {
-			// Print-error as a warning.
-			// This is a non-fatal error because object was successfully applied
-			// above, but it might have issues since migration failed.
-			//
-			// This migration will be re-attempted if necessary upon next
-			// apply.
-			fmt.Fprintf(o.ErrOut, warningMigrationPatchFailed, err.Error())
-		} else if performedMigration {
-			if obj, err = helper.Patch(
-				info.Namespace,
-				info.Name,
-				types.ApplyPatchType,
-				data,
-				&options,
-			); err != nil {
-				// Re-send original SSA patch (this will allow dropped fields to
-				// finally be removed)
-				fmt.Fprintf(o.ErrOut, warningMigrationReapplyFailed, err.Error())
-			} else {
-				info.Refresh(obj, false)
-			}
-		}
-
-		WarnIfDeleting(info.Object, o.ErrOut)
-
-		if err := o.MarkObjectVisited(info); err != nil {
-			return err
-		}
+	if o.ServerSideApply == "true" {
+		return o.serverSideApplyOneObject(info, helper)
+	}
 
-		if o.shouldPrintObject() {
-			return nil
-		}
+	if o.ServerSideApply == "auto" {
+		// Check last_applied
+		metadata, _ := meta.Accessor(info.Object)
+		annotationMap := metadata.GetAnnotations()
+		_, hasAnnotation := annotationMap[corev1.LastAppliedConfigAnnotation]
+		fmt.Printf("HAS ANNOTATION: %s", hasAnnotation)
 
-		printer, err := o.ToPrinter("serverside-applied")
-		if err != nil {
-			return err
+		if !hasAnnotation {
+			fmt.Println("I am going to serverside apply")
+			return o.serverSideApplyOneObject(info, helper)
```

Comment:
Note this is still not in the correct place - I am stilling thinking/working out the best way to check the annotation without adding additional requests.